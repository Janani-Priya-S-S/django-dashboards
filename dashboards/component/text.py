from dataclasses import dataclass

from .base import Component
from .stat import *


@dataclass
class Text(Component):
    template_name: str = "dashboards/components/text/text.html"
    mark_safe: bool = False
