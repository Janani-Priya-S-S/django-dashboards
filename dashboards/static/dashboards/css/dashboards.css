:root {
  --color-background-primary: #e5e5e5;
  --color-background-secondary: #fafafa;
  --color-background-tertiary: #f5f5f5;
  --color-border: #d4d4d4;
  --color-text-primary: #404040;
  --color-text-secondary: #a3a3a3;
  --shadow: 0 0 0 0px #fff, 0 20px 25px -5px rgba(0,0,0,0.10196), 0 8px 10px -6px rgba(0,0,0,0.10196);
  --root-background: linear-gradient(to top right, var(--color-background-primary), var(--color-background-primary), var(--color-border));
  --color-danger: #dc2626;
  --color-cta-background: #737373;
  --color-cta-hover-background: #404040;
  --color-cta-text: #fafafa;
  --color-cta-hover-text: var(--color-cta-text);
  --color-cta-alt-background: #a3a3a3;
  --color-cta-alt-hover-background: #fafafa;
  --color-cta-alt-text: #404040;
  --color-cta-alt-hover-text: var(--color-cta-alt-text);
  --transition-time: 0.25s;
}

@media (prefers-color-scheme: dark) {
  :root {
    --color-background-primary: #404040;
    --color-background-secondary: #525252;
    --color-background-tertiary: #737373;
    --color-border: #737373;
    --color-text-primary: #fafafa;
    --color-text-secondary: #a3a3a3;
    --root-background: linear-gradient(to top right, var(--color-background-primary), var(--color-background-primary), var(--color-background-secondary));
    --color-danger: #fca5a5;
    --color-cta-background: #a3a3a3;
    --color-cta-hover-background: #fafafa;
    --color-cta-text: #404040;
    --color-cta-hover-text: var(--color-cta-text);
    --color-cta-alt-background: #737373;
    --color-cta-alt-hover-background: #404040;
    --color-cta-alt-text: #fafafa;
    --color-cta-alt-hover-text: var(--color-cta-alt-text);
  }
}

.btn {
  height: 2em;
  border-radius: 1em;
  text-align: center;
  border: 0;
  box-shadow: var(--shadow);
  transition: all var(--transition-time);
  padding: 0 1em;
}

.btn.cta {
  background-color: var(--color-cta-background);
  color: var(--color-cta-text);
}

.btn.cta:hover,
.btn.cta:active {
  background-color: var(--color-cta-hover-background);
  color: var(--color-cta-hover-text);
}

.btn.cta.alt {
  background-color: var(--color-cta-alt-background);
  color: var(--color-cta-alt-text);
}

.btn.cta.alt:hover,
.btn.cta.alt:active {
  background-color: var(--color-cta-alt-hover-background);
  color: var(--color-cta-alt-hover-text);
}

.card {
  position: relative;
  height: 100%;
}

.card .card-header {
  display: flex;
  justify-content: space-between;
  margin-bottom: 0;
  background-color: var(--color-background-tertiary);
  padding: 1rem;
  border-bottom: solid 1px rgba(154, 161, 171, 0.15);
  border-bottom: solid 1px var(--color-border);
}

.card .card-header .header-title {
  flex: 1 1 auto;
  font-size: 1.2em;
}

.card .card-header .dropdown {
  position: relative;
  display: inline-block;
}

.card .card-header .dropdown .dropdown-btn {
  color: white;
  padding: 6px;
  font-size: 1em;
  border: none;
  cursor: pointer;
}

.card .card-header .dropdown .dropdown-content {
  position: absolute;
  min-width: 160px;
  box-shadow: 0px 8px 16px 0px rgba(0, 0, 0, 0.2);
  z-index: 1;
}

.card .card-header .dropdown .dropdown-content a {
  color: black;
  font-size: 1em;
  padding: 12px 16px;
  -webkit-text-decoration: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: block;
}

.card .card-header .dropdown .dropdown-content a:hover {
  background-color: #f1f1f1;
}

.card .card-img {
  width: 100%;
}

.card .card-body {
  padding: 5px;
}

.card-body-grid {
  display: grid;
  grid-template-columns: repeat(12, 1fr);
  grid-column-gap: 10px;
}

.card .card-body p {
  font-size: 1rem;
}

.card .card-footer {
  flex: 1 1 auto;
  padding: 5px;
  font-size: 0.75em;
  background-color: var(--color-background-tertiary);
}

.form-table {
  text-align: left;
}

.form-table label {
  font-weight: bold;
}

.form-table input,
.form-table textarea,
.form-table select {
  padding: 0.5em;
  border-radius: 0.5em;
  width: 100%;
  border: 1px solid var(--border-color);
  background: var(--color-background-primary);
  color: var(--color-text-primary)
}

.form-table .helptext {
  color: var(--color-text-secondary);
  margin: 0;
  font-size: 0.7rem;
}

.errorlist {
  color: var(--color-danger);
  list-style: none;
  text-align: left;
  padding-left: 10px;
  margin: 0;
}

.form-table .errorlist {
  padding-left: 0;
}

* {
  box-sizing: border-box;
  text-rendering: optimizeLegibility;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-kerning: auto;
}

html {
  font-size: 1.2em;
  line-height: 1.4;
  font-weight: 400;
  font-family: Inter var,ui-sans-serif,system-ui, -apple-system, Segoe UI, Roboto, Ubuntu, Cantarell, Noto Sans, sans-serif,BlinkMacSystemFont,Segoe UI,Roboto,Helvetica Neue,Arial,Noto Sans,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol,Noto Color Emoji;
  background: var(--root-background);
  color: var(--color-text-primary);
}

body {
  padding: 1em;
  margin: 0 auto;
  max-width: 95%;
  min-height: 100vh;
}

a{
  color: var(--color-text-primary);
  cursor: pointer;
  font-size: 1.1rem;
  -webkit-text-decoration: none;
  text-decoration: none;
}

h1, h2, h3, h4, h5, h6 {
  margin: 0 0 .5em 0;
  line-height: 1.2;
  letter-spacing: -.02em;
}

hr{
  border-color: var(--color-text-primary);
}

.menu {
  background: var(--color-background-secondary);
  list-style: none;
  margin: 0;
  padding-left: 0;
  display: flex;
  z-index: 2;
}

.menu li {
  background: var(--color-background-secondary);
  display: block;
  float: left;
  padding: 0.5rem 2rem;
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.menu li a {
  color: var(--color-text-primary);
}

.menu li.active, .menu li ul li.active {
  background-color: var(--color-background-tertiary);
}

.menu li:hover,
.menu li:focus-within {
  background-color: var(--color-background-tertiary);
  cursor: pointer;
}

.menu li:focus-within a {
  outline: none;
}

.menu .dropdown {
  display: none;
  left: 0;
  margin: 0.5rem 0 0;
  padding: 0;
  min-width: 4em;
  opacity: 0;
  position: absolute;
  visibility: hidden;
  background: var(--color-background-tertiary);
  border: 1px solid var(--color-border);
  border-top: 0;
  overflow: hidden;
  border-radius: 0 0 15px 15px;
}

.menu li:hover > .dropdown,
.menu li:focus-within > .dropdown,
.menu li .dropdown:hover,
.menu li .dropdown:focus {
  visibility: visible;
  opacity: 1;
  display: block;
  z-index: 1;
}

.menu .dropdown li {
  clear: both;
  width: 100%;
  padding: 0;
}

.menu .dropdown li a {
  display: block;
  padding: 0.5rem 2rem;
}

.menu-card {
  display: flex;
  margin-bottom: 1em;
}

.menu-card nav {
  flex-grow: 1;
  box-shadow: var(--shadow);
  border: 1px solid var(--color-border);
  border-right: 0;
  border-left: 0;
}

.menu-card .bumper {
  width: 15px;
  overflow: hidden;
  background: var(--color-background-secondary);
  border: 1px solid var(--color-border);
  z-index: 1;
}

.menu-card .bumper:first-child {
  border-radius: 15px 0 0 15px;
  border-right: 0;
}

.menu-card .bumper:last-child {
  border-radius: 0 15px 15px 0;
  border-left: 0;
}

/* This is an example - you'd ideally use a lib or scss to compile this, add responsiveness etc */

.span-1 {
  grid-column-end: span 1 !important;
}

.span-1 {
  grid-auto-rows: min-content;
}

.span-2 {
  grid-column-end: span 2 !important;
}

.span-2 {
  grid-auto-rows: min-content;
}

.span-3 {
  grid-column-end: span 3 !important;
}

.span-3 {
  grid-auto-rows: min-content;
}

.span-4 {
  grid-column-end: span 4 !important;
}

.span-4 {
  grid-auto-rows: min-content;
}

.span-5 {
  grid-column-end: span 5 !important;
}

.span-5 {
  grid-auto-rows: min-content;
}

.span-6 {
  grid-column-end: span 6;
  grid-auto-rows: min-content;
}

.span-7 {
  grid-column-end: span 7;
  grid-auto-rows: min-content;
}

.span-8 {
  grid-column-end: span 8;
  grid-auto-rows: min-content;
}

.span-9 {
  grid-column-end: span 9;
  grid-auto-rows: min-content;
}

.span-10 {
  grid-column-end: span 10;
  grid-auto-rows: min-content;
}

.span-11 {
  grid-column-end: span 11;
  grid-auto-rows: min-content;
}

.span-12 {
  grid-column-end: span 12;
  grid-auto-rows: min-content;
}

.dashboard-container {
  display: grid;
  grid-template-columns: repeat(12, 1fr);
  grid-gap: 1rem;
  gap: 1rem;
  text-align: center;
  width: 100%;
}

h1, h2, h3, h4, p, .tab-container {
  grid-column: 1 / 13;
}

.dashboard-component {
  border-radius: 15px;
  background-color: var(--color-background-secondary);
  display: grid;
  grid-template-columns: repeat(12, 1fr);
  box-shadow: var(--shadow);
  border: 1px solid var(--color-border);
}

.dashboard-component * {
  /* allow sub nesting, by assuming not set is 12 */
  grid-column-end: span 12;
}

.tab-content div.active .dashboard-component {
  border-radius: 0 15px 15px 15px;
}

.dashboard-component-inner {
  padding: 2rem;
}

.dashboard-component-cta {
  color: var(--color-text-primary);
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1.3rem;
}

.tabs {
  border-top: 2px var(--color-border) solid;
  display: flex;
  flex-wrap: wrap;
  margin-left: 0;
  padding: 35px 0 0;
  list-style: none;
}

.tabs div, .tabs .tab {
  background-color: var(--color-background-secondary);
  padding: 10px 20px;
}

.tabs div a, .tabs .tab a {
  color: var(--color-text-primary);
}

.tabs div a.active, .tabs .tab a.active {
  color: var(--color-text-primary);
  border-bottom: 1px solid var(--color-background-secondary);
}

.dataTables_filter, .dataTables_length, .dataTables_info, .dataTables_paginate {
  color: var(--color-text-primary) !important;
  font-size: 0.9rem !important;
}

.dataTables_filter, .dataTables_length, .dataTables_info, .dataTables_paginate {
  margin-bottom: 10px;
}

.dataTables_length select {
  color: var(--color-text-primary) !important;
}

.table {
  border-collapse: collapse;
  border-spacing: 0;
  width: 100%;
  font-size: 0.8rem;
}

.table thead th {
  background-color: var(--color-background-secondary);
  padding: 1rem;
  text-align: left;
  text-shadow: 1px 1px 1px var(--color-border);
}

.table tbody td {
  border: solid 1px var(--color-background-secondary);
  padding: 10px;
}

.customComponent {
  background-color: var(--color-background-primary);
  border-radius: 20px;
  padding: 10px 5px;
  margin: 10px;
  font-size: 0.8em;
}

.gauge-container {
  margin: auto;
  max-width: 65%;
}
