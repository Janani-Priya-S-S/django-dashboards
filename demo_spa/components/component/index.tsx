export * from "./charts"
export * from "./html"
export * from "./table"
export * from "./text"
export * from "./display"