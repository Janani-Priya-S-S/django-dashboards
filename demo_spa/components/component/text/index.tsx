import React from "react";

export const Text: React.FC<{value: string}> = ({value}) => (
  <div>{value}</div>
)

export const HTML: React.FC<{value: string}> = ({value}) => (
  <div dangerouslySetInnerHTML={{ __html: value }}/>
)

export const Stat: React.FC<{value: {text: string, sub_text: string}}> = ({value}) => (
    <div>
        <h2>{value.text}</h2>
        <small>{value.sub_text}</small>
    </div>
)