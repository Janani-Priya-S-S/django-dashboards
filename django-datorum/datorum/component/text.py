from dataclasses import dataclass
from typing import Optional

from .base import Component


@dataclass
class Text(Component):
    template: str = "datorum/components/text/text.html"
    mark_safe: bool = False


@dataclass
class StatData:
    text: str
    change_by: Optional[float] = None
    sub_text: Optional[str] = ""


@dataclass
class Stat(Component):
    template: str = "datorum/components/text/stat.html"
    href: Optional[str] = None


@dataclass
class CTAData:
    text: str
    href: str


@dataclass
class CTA(Component):
    template: str = "datorum/components/cta.html"
    value: Optional[CTAData] = None


@dataclass
class ProgressData:
    @dataclass
    class ProgressItem:
        value: str
        percentage: int
        title: Optional[str] = None

    data: list[ProgressItem]


@dataclass
class Progress(Component):
    template: str = "datorum/components/text/progress.html"
