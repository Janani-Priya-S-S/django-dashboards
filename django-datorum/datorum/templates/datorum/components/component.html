{% if component.is_deferred and htmx %}
    <div hx-get="{{ component.get_absolute_url }}?key={{ component.key }}" hx-trigger="intersect once"></div>
{% else %}
    {% include "datorum/components/filter.html" %}

    {# -inner will be replaced if a filter is applied. #}
    <div id="component-{{ component.key }}-inner" class="dashboard-component-inner fade-in">
        {% include component.template %}
        {% if component.cta %}
          <div>
            {% include component.cta.template with component=component.cta %}
          </div>
        {% endif %}
    </div>
{% endif %}