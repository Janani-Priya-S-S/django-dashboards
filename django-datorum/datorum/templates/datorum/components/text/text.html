{% load datorum %}
{% render component as text %}
{{ text }}
