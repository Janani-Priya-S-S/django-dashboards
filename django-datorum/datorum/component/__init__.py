from .base import Component
from .chart import Chart
from .form import Form
from .map import Map
from .table import BasicTable, Table
from .text import CTA, Stat, Text


__all__ = [
    "Component",
    "CTA",
    "Chart",
    "Form",
    "Map",
    "Stat",
    "Text",
    "Table",
    "BasicTable",
]
