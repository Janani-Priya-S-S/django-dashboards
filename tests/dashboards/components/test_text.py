import datetime

from django.template import Context

import pytest

from tests.utils import render_component_test
from wildcoeus.dashboards.component import CTA
from wildcoeus.dashboards.component.text import (
    CTAData,
    Progress,
    ProgressData,
    Stat,
    StatData,
    TimelineData,
)


pytest_plugins = [
    "tests.dashboards.fixtures",
]


@pytest.mark.parametrize("htmx", [True, False])
def test_cta_component__renders_value(htmx, dashboard, rf, snapshot):
    component = CTA(value=CTAData(text="click here", href="/"))
    component.key = "test"
    component.dashboard = dashboard
    context = Context(
        {
            "component": component,
            "request": rf.get("/"),
        }
    )

    snapshot.assert_match(render_component_test(context, htmx=htmx))


@pytest.mark.parametrize(
    "component_kwargs",
    [
        {"value": StatData(text="100%", sub_text="increase")},
        {"defer": lambda **k: StatData(text="100%", sub_text="increase")},
        {
            "value": StatData(
                text="100%", sub_text="increase", change_by=1.0, change_by_text="Change"
            )
        },
    ],
)
@pytest.mark.parametrize("htmx", [True, False])
def test_stat_component__renders_value(component_kwargs, dashboard, htmx, rf, snapshot):
    component = Stat(**component_kwargs)
    component.dashboard = dashboard
    component.key = "test"
    context = Context(
        {
            "component": component,
            "request": rf.get("/"),
        }
    )

    snapshot.assert_match(render_component_test(context, htmx=htmx))


@pytest.mark.parametrize(
    "component_kwargs",
    [
        {
            "value": ProgressData(
                data=[
                    ProgressData.ProgressItem(value="100%", percentage=100, title=None)
                ]
            )
        },
        {
            "defer": lambda **k: ProgressData(
                data=[
                    ProgressData.ProgressItem(value="100%", percentage=100, title=None)
                ]
            )
        },
        {
            "value": ProgressData(
                data=[
                    ProgressData.ProgressItem(value="100%", percentage=100, title="a"),
                    ProgressData.ProgressItem(value="100%", percentage=100, title="b"),
                ]
            )
        },
    ],
)
@pytest.mark.parametrize("htmx", [True, False])
def test_progress_component__renders_value(
    component_kwargs, dashboard, htmx, rf, snapshot
):
    component = Progress(**component_kwargs)
    component.dashboard = dashboard
    component.key = "test"
    context = Context(
        {
            "component": component,
            "request": rf.get("/"),
        }
    )

    snapshot.assert_match(render_component_test(context, htmx=htmx))


@pytest.mark.parametrize(
    "component_kwargs",
    [
        {
            "value": TimelineData(
                items=[
                    TimelineData.TimelineItem(
                        icon="<>",
                        title="t1",
                        subtext="st1",
                        datetime=datetime.datetime(2020, 1, 1),
                    )
                ]
            ),
        },
        {
            "defer": lambda **k: TimelineData(
                items=[
                    TimelineData.TimelineItem(
                        icon="<>",
                        title="t1",
                        subtext="st1",
                        datetime=datetime.datetime(2020, 1, 1),
                    )
                ]
            )
        },
        {
            "value": TimelineData(
                items=[
                    TimelineData.TimelineItem(
                        icon="<>",
                        title="t1",
                        subtext="st1",
                        datetime=datetime.datetime(2020, 1, 1),
                    ),
                    TimelineData.TimelineItem(
                        icon="<>",
                        title="t2",
                        subtext="st2",
                        datetime=datetime.datetime(2020, 1, 1),
                    ),
                ]
            ),
        },
    ],
)
@pytest.mark.parametrize("htmx", [True, False])
def test_timeline_component__renders_value(
    component_kwargs, dashboard, htmx, rf, snapshot
):
    component = Progress(**component_kwargs)
    component.dashboard = dashboard
    component.key = "test"
    context = Context(
        {
            "component": component,
            "request": rf.get("/"),
        }
    )

    snapshot.assert_match(render_component_test(context, htmx=htmx))
