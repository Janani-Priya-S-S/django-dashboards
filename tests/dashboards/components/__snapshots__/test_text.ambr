# serializer version: 1
# name: test_cta_component__renders_value[False]
  '''
  
  
  
      <a href="/">
          <div id="component-dashapp1testdashboardcomponenttest-inner" class="dashboard-component-inner fade-in">
              
  click here
  
          </div>
      </a>
  
  
  
  '''
# ---
# name: test_cta_component__renders_value[True]
  '''
  
  
  
      <a href="/">
          <div id="component-dashapp1testdashboardcomponenttest-inner" class="dashboard-component-inner fade-in">
              
  click here
  
          </div>
      </a>
  
  
  
  '''
# ---
